lib/gems/pending/appliance_console/prompts.rb

```diff
@@ -162,6 +162,7 @@ def ask_for_disk(disk_name, verify = true)
 
       if verify && disk.nil?
         say ""
+        raise MiqSignalError unless allow_none
```

Comment:
I feel like this won't work if the method is called without verify.

So as an example `ask_for_disk("disk", false, false)` would not raise.

Also, we have a check for `disks.empty?` above, should this raise be there rather than down here?

Reply:
I thought verify is to ask manageiq to check disk. verify=false is only used in case we're sure disk is valid or invalid disk doesn't matter, in either case no need to verify. So allow_none only make sense and should be used when verify =true. But your comment seems better. I'll change it.